apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -0,0 +1,103 @@
+}
+
+(function(exports) {
+  var service;
+
+  const SERVICE_NAME = 'conversation-service';
+
+  const STREAMS = Object.freeze(
+    ['getAllConversations', 'getMessagesForConversation']
+  );
+
+  const METHODS = Object.freeze([
+    'deleteConversations', 'deleteMessages', 'markConversationsAs',
+    'getConversationSummary', 'getMessage', 'findConversationFromAddress'
+  ]);
+
+  const EVENTS = Object.freeze([
+    'message-change'
+  ]);
+
+  var ConversationService = {
+    /**
+     * Initializes our service using the Service mixin's initService.
+     */
+    init() {
+      service = this.initService();
+    },
+
+    /**
+     * Stream that returns everything needed to display conversations in the
+     * inbox view.
+     * @returns {Stream.<ConversationSummary>}
+     */
+    getAllConversations(stream) {},
+
+    /**
+     * Stream that returns messages in a conversation, with all suitable
+     * information to display in the conversation view.
+     * @param {Number} conversationId Conversation to retrieve.
+     * @returns {Stream.<MessageSummary>}
+     */
+    getMessagesForConversation(stream, conversationId) {},
+
+    /**
+     * Returns all information related to a single conversation, to display in
+     * the conversation view.
+     * @param {Number} conversationId Conversation to retrieve.
+     */
+    getConversationSummary(conversationId) {},
+
+    /**
+     * Deletes one or several conversations, including their messages.
+     * @param {Array.<Number>} conversationIds ID of the conversations to be
+     * deleted.
+     */
+    deleteConversations(conversationIds) {},
+
+    /**
+     * Deletes one or several messages.
+     * @param {Array.<Number>} messageIds ID of the messages to be deleted.
+     */
+    deleteMessages(messageIds) {},
+
+    /**
+     * Marks a conversation as read or unread.
+     * @param {Object} param
+     * @param {Array.<Number>} param.conversations Conversation IDs to be
+     * marked.
+     * @param {Boolean} param.asRead Specifies whether we mark as read of
+     * unread.
+     */
+    markConversationsAs({conversations, asRead}) {},
+
+    /**
+     * Fetches a message information.
+     * @param {Number} messageId The message ID to retrieve.
+     * @returns {Message} All message information, including contact, delivery,
+     * read reports, size, timestamps, etc.
+     */
+    getMessage(messageId) {},
+
+    /**
+     * Finds a conversation from an address (that is: a phone number or an
+     * email).
+     * @param {String} address This is either a phone number or an email, to
+     * match an existing conversation from.
+     * @returns {Number?} The id of the conversation, or null if not found.
+     */
+    findConversationFromAddress(address) {}
+  };
+
+  exports.ConversationService = Object.freeze(
+    BridgeServiceMixin.mixin(
+      ConversationService,
+      { name: SERVICE_NAME, methods: METHODS, streams: STREAMS, events: EVENTS }
```

Comment:
Just asking, would not it better and clearer if we pass ```name``` as separate required parameter?